features/helper_specs/helper_spec.feature

```diff
@@ -120,3 +120,96 @@ Feature: helper spec
       """
     When I run `rspec spec/helpers/widgets_helper_spec.rb`
     Then the examples should all pass
+
+  Scenario: helper methods are accessible from within example
+    Given a file named "spec/helpers/user_helper_spec.rb" with:
+      """ruby
+      require "rails_helper"
+
+      RSpec.describe UserHelper, :type => :helper do
+        describe "#email" do
+          it "gives current user email" do
+            expect(email).to eq "jon.snow@example.com"
+          end
+        end
+      end
+      """
+    And a file named "app/helpers/user_helper.rb" with:
+      """ruby
+      module UserHelper
+        def email
+          "jon.snow@example.com"
+        end
+      end
+      """
+    When I run `rspec spec/helpers/user_helper_spec.rb`
+    Then the examples should all pass
+
+  Scenario: methods defined in example are visible within example
+    Given a file named "spec/helpers/user_helper_spec.rb" with:
+      """ruby
+      require "rails_helper"
+
+      RSpec.describe UserHelper, :type => :helper do
+        describe "#email" do
+          context "when user is logged in" do
+            let(:current_user) { double("user", {:email => "jon.snow@example.com"}) }
+
+            it "gives current user email" do
+              expect(email).to eq "jon.snow@example.com"
+            end
+          end
+
+          context "when user isn't logged in" do
+            def current_user; end
+
+            it "gives n/a" do
+              expect(email).to eq "n/a"
+            end
+          end
+
+          context "when current_user isn't defined" do
+            it "throws an exception" do
+              expect { email }.to raise_error NameError
+            end
+          end
+        end
+      end
+      """
+    And a file named "app/helpers/user_helper.rb" with:
+      """ruby
+      module UserHelper
+        def email
+          current_user.nil? ? 'n/a' : current_user.email
+        end
+      end
+      """
+    When I run `rspec spec/helpers/user_helper_spec.rb`
+    Then the examples should all pass
+
+  Scenario: methods defined in example are not visible in helper
+    Given a file named "spec/helpers/user_helper_spec.rb" with:
+      """ruby
+      require "rails_helper"
+
+      RSpec.describe UserHelper, :type => :helper do
+        describe "#email" do
+          context "when current_user is defined" do
+            let(:current_user) { double("user", {:email => "jon.snow@example.com"}) }
+            it "throws an exception" do
+              expect { helper.email }.to raise_error NameError, /undefined local variable or method `current_user'/
+            end
+          end
+        end
+      end
+      """
+    And a file named "app/helpers/user_helper.rb" with:
+      """ruby
+      module UserHelper
+        def email
+          current_user.email
+        end
+      end
+      """
+    When I run `rspec spec/helpers/user_helper_spec.rb`
+    Then the examples should all pass
```

Comment:
Given the surprising behavior illustrated above I think this is a spec we will need to add. I'm not sure that this level of the docs is the proper place; as I would never expect this to work. I think this type of check should be belong in the regular spec suite. If for some reason that's not possible then the internal acceptance suite would be a good place.